pwiz_tools/Skyline/Model/ModificationMatcher.cs

```diff
@@ -353,6 +353,32 @@ public static StaticMod GetStaticMod(string unimodNameOrId, ModTerminus? modTerm
                 };
                 if (UniMod.DictUniModIds.TryGetValue(key, out var mod))
                     return mod;
+
+                // if both AA and terminus are provided, look up keys with one or the other set instead of both

```

Comment:
Was there a particular unimod modification that wasn't getting matched correctly that necessitated this change?
If so, then I imagine it was probably also a problem if the user tried to add the modified peptide to the document using "Edit > Insert > Peptides".

Reply:
This code was originally added for the command-line modification parameters, and this change will only help that case by making it more tolerant of the user being TOO specific. For example, if they do: "--pep-unimod-add 4 --pep-unimod-aa C --pep-unimod-term N" it would add "Carbamidomethyl (C)". I had to add this for the new ModifiedSequence constructor so it would parse "C[UNIMOD:4]PEPTIDEK" correctly (because that C is both N-terminal and C).

Reply:
Maybe we should add Nick's new code to prefer "common" modifications over not common modifications.

https://github.com/ProteoWizard/pwiz/pull/2859

I worry here that we are preferring AA mods over terminal mods, which proved to be the wrong choice for "Acetyl" which has many AA versions and also an N-terminal version which is common. In a tutorial, we ended up choosing lots of AA mods instead of Acetyl (N-term) which covered everything with a single modification.

Reply:
Oops. I didn't realize that "GetStaticMod" never actually gets used by the class "ModificationMatcher"-- it's only used by CommandLine.

"ModificationMatcher" is a nightmare of a class which tries to guess what modifications a user intended based on whatever they typed in. It was written by brilliant interns decades ago and we are terrified of changing anything about it.

This "GetStaticMod" method probably should be moved into the class "UniMod" and should maybe be renamed to something like "FindById".